akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/EventSourcedBehaviorTestKitImpl.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.testkit.internal
+
+import scala.concurrent.duration._
+
+import akka.actor.testkit.typed.scaladsl.ActorTestKit
+import akka.actor.testkit.typed.scaladsl.TestInbox
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.persistence.testkit.scaladsl.EventSourcedBehaviorTestKit
+import akka.persistence.testkit.scaladsl.PersistenceTestKit
+import akka.persistence.typed.PersistenceId
+import akka.persistence.typed.internal.EventSourcedBehaviorImpl
+
+class EventSourcedBehaviorTestKitImpl[Command, Event, State](actorTestKit: ActorTestKit, behavior: Behavior[Command])
+    extends EventSourcedBehaviorTestKit[Command, Event, State] {
+  import EventSourcedBehaviorTestKit.CommandResult
+
+  private val persistenceTestKit = PersistenceTestKit(actorTestKit.system)
+  private val probe = actorTestKit.createTestProbe[Any]()
+  private val actor: ActorRef[Command] = actorTestKit.spawn(behavior)
+  private val internalActor = actor.unsafeUpcast[Any]
+  internalActor ! EventSourcedBehaviorImpl.GetPersistenceId(probe.ref)
+  private val persistenceId: PersistenceId = probe.expectMessageType[PersistenceId]
+
+  persistenceTestKit.clearByPersistenceId(persistenceId.id)
+
+  override def runCommand(command: Command): CommandResult[Command, Event, State, Nothing] = {
+    // FIXME we can expand the api of persistenceTestKit to read from storage from a seqNr instead
+    val oldEvents =
+      persistenceTestKit.persistedInStorage(persistenceId.id).map(_.asInstanceOf[Event])
+
+    actor ! command
+
+    // FIXME signals are not stashed, maybe they should (and then this awaitAssert wouldn't be needed
+    val tmpStateProbe = actorTestKit.createTestProbe[State]()
+    Thread.sleep(50)
```

Comment:
why sleep before an awaitAssert?

Reply:
Was just an idea for optimization, but I have some work in progress removing the awaitAssert altogether.

Reply:
updated now